updatecli/policies/open-release-pr/updatecli.d/open-release-pr.yaml

```diff
@@ -0,0 +1,164 @@
+    kind: githubrelease
+    spec:
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repo }}"
+      token: "{{ requiredEnv .github.token }}"
+      key: "title"
+      versionfilter:
+        kind: latest
+
+  cargoPackageName:
+    name: Extract cargo package name from Cargo.toml
+    kind: toml
+    scmid: "policy-repo"
+    dependson:
+      - condition#checkCargoTomlExists
+    spec:
+      file: "Cargo.toml"
+      key: "package.name"
+
+conditions:
+  checkCargoTomlExists:
+    name: Check if Cargo.toml exists
+    scmid: "policy-repo"
+    kind: file
+    disablesourceinput: true
+    spec:
+      file: Cargo.toml
+  checkCargoLockExists:
+    name: Check if Cargo.lock exists
+    scmid: "policy-repo"
+    kind: file
+    disablesourceinput: true
+    spec:
+      file: Cargo.lock
+
+targets:
+  updateMetadataAnnotationVersion:
+    name: update metadata.yml with new version
+    kind: yaml
+    scmid: "policy-repo"
+    sourceid: newestGitHubDraftReleaseVersion
+    spec:
+      searchpattern: true
+      file: "metadata.y*ml"
+      key: "$.annotations.'io.kubewarden.policy.version'"
+      # value: '{{ source  "newestGitHubDraftReleaseVersion" }}'
+
+  updateCargoTomlVersion:
+    name: update Cargo.toml with new version
+    scmid: "policy-repo"
+    disablesourceinput: true
+    dependson:
+      - condition#checkCargoTomlExists
+    kind: shell
+    spec:
+      shell: /usr/bin/bash
+      command: |
+        export new_version='{{ source "newestGitHubDraftReleaseVersion" }}'
+        awk -v new_version="$new_version" '
+          BEGIN { in_package = 0 }
+          /^\[package\]/ { in_package = 1 }
+          in_package && /^version = / { sub(/version = ".*"/, "version = \"" new_version "\"") }
+          /^\[/ && !/^\[package\]/ { in_package = 0 }
+          { print }
+        ' Cargo.toml > Cargo.toml.tmp && mv Cargo.toml.tmp Cargo.toml
+      changedif:
+        kind: file/checksum
+        spec:
+          files:
+            - Cargo.toml
+
+  updateCargoLockVersion:
+    name: update Cargo.lock with new version
+    scmid: "policy-repo"
+    disablesourceinput: true
+    dependson:
+      - condition#checkCargoLockExists
+      - target#updateCargoTomlVersion
+    kind: shell
+    spec:
+      shell: /usr/bin/bash
+      command: |
+        export package_name='{{ source "cargoPackageName" }}'
+        export new_version='{{ source "newestGitHubDraftReleaseVersion" }}'
+        # note: if testing locally, copy the awk command without trailing spaces
+        awk -v new_version="$new_version" -v package_name="$package_name" '
+          BEGIN { package_found = 0 }
+          /^\[\[package\]\]/ { package_found = 0 }
+          $0 ~ "^name = \"" package_name "\"$" { package_found = 1 }
+          package_found && /^\s*version = / { sub(/version = ".*"/, "version = \"" new_version "\"") }
+          { print }
+        ' Cargo.lock > Cargo.lock.tmp && mv Cargo.lock.tmp Cargo.lock
```

Comment:
I would just run `cargo check`

Reply:
In principle it would be a good idea, but then we need `cargo` installed, which means having several reusable workflows per language. So while clunky I think this is best.

Reply:
You could run a condition if cargo is installed then run the target otherwise skip

Reply:
I'm not familiar with how the updatecli is being run inside of the GHA. Isn't it just running inside of the usual ubuntu base image used by the runner? I though the updatecli GHA simply installs the go statically linked binary into the PATH

Reply:
It depends how the GitHub action is configured but the updatecli github action just install Updatecli on the runner so other tools are available

Reply:
yes, it would be possible. The reusable workflow that would use the updatecli policy could install cargo conditionally if we find a Cargo.toml file.

Reply:
Done in d4a7728.